source/CCustomOpcodeSystem.cpp

```diff
@@ -1361,26 +1361,30 @@ namespace CLEO
 	OpcodeResult __stdcall opcode_0A8D(CRunningScript *thread)
 	{
 		GetScriptParams(thread, 3);
-		//DWORD value;
 		void *Address = opcodeParams[0].pParam;
 		DWORD size = opcodeParams[1].dwParam;
 		bool vp = opcodeParams[2].bParam;
 
 		opcodeParams[0].dwParam = 0;
-
 		switch (size)
 		{
+		case 0:
+			break;

```

Comment:
should be a warning I think, i.e. default case

Reply:
fixed

Reply:
I don't get the current logic. should be something like

```cpp
		switch (size)
		{
		case 1:
			GetInstance().CodeInjector.MemoryRead(Address, (BYTE)opcodeParams[0].ucParam, vp);
			break;
		case 2:
			GetInstance().CodeInjector.MemoryRead(Address, (WORD)opcodeParams[0].usParam, vp);
			break;
		case 3:
			GetInstance().CodeInjector.MemoryRead(Address, (DWORD)opcodeParams[0].dwParam, vp);
			opcodeParams[0].dwParam &= 0x00FFFFFF;
			break;
		case 4:
			GetInstance().CodeInjector.MemoryRead(Address, (DWORD)opcodeParams[0].dwParam, vp);
			break;
		}
		default:
			LOG_WARNING("Invalid size param (%d) of opcode [0A8D] in script %s", size, ((CCustomScript*)thread)->GetInfoStr().c_str());
```